crates/http/src/backend/h3/mod.rs

```diff
@@ -0,0 +1,224 @@
+use std::fmt::Debug;
+use std::io;
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use body::QuicIncomingBody;
+use scuffle_context::ContextFutExt;
+#[cfg(feature = "tracing")]
+use tracing::Instrument;
+use utils::copy_response_body;
+
+use crate::error::Error;
+use crate::service::{HttpService, HttpServiceFactory};
+
+pub mod body;
+mod utils;
+
+/// A backend that handles incoming HTTP3 connections.
+///
+/// This is used internally by the [`HttpServer`](crate::server::HttpServer) but can be used directly if preferred.
+///
+/// Call [`run`](Http3Backend::run) to start the server.
+#[derive(bon::Builder, Debug, Clone)]
```

Comment:
We should add comments to each field so that the builder displays the comments on docs.